### PR TITLE
Adds an example for stacked monad transformers

### PR DESCRIPTION
**Pull request by NickDarvey**
This example is based on @gusty answer to my question in https://github.com/fsprojects/FSharpPlus/issues/301.

It builds on the previous example in the doc showing how you can add one more layer.

### CODE REVIEW

File: docsrc/content/abstraction-monad.fsx [hunk 1]
```diff
@@ -219,6 +219,42 @@ module CombineWriterWithResult =
...
 
 
+// You can also stack monad transformers.
+
+// A monad transformer and a monad is itself a monad, so you can pass that into another monad transformer.
+// For example, below we are stacking them like:
+// type Example = ReaderT<DateTime, ResultT<Writer<string list, Result<string * string * string, string>>>>)
+
+// Catch and throw is generic over all monad transformers in F#+ so you can use catch works in this example
+// because there is a Result in the stack. We use it here to consolidate Result's 'TError.
```
Comment (mitimes-engineering) [inline docsrc/content/abstraction-monad.fsx]: This sentence doesn't read quite right...

Comment (adz) [reply docsrc/content/abstraction-monad.fsx]: Oops, I was logged in a work service account. Was me in above ! :)

File: docsrc/content/abstraction-monad.fsx [hunk 2]
```diff
@@ -219,6 +219,42 @@ module CombineWriterWithResult =
...
+
+    let ew = monad {
+        let! w = eitherConv divide5By       6.0
+        let! x = eitherConv divide5By       3.0
+        let! y = eitherConv divide5By       0.0
+        let! z = eitherConv otherDivide5By  0.0 </catch/> (throw << (fun _ -> "Unknown error"))
+
+        return (x, y, z) }
+
+    let (_, log) = ew |> ReaderT.run >> ResultT.run >> Writer.run <| DateTime.UtcNow
```
Comment (adz) [inline docsrc/content/abstraction-monad.fsx]: Not sure about using `ew` here in docs.
Is it meaning "either_and_writer" (and ignore the reader bit)?

Also, maybe this should be broken down a little to show the intermediary parts?
It's hard to decipher `|>` `>>` `>>` and `<|` in one line, even though it is quite neat.

**Comment by adz** [comment]
Nice example of the stack there! Thanks.

File: docsrc/content/abstraction-monad.fsx (hunk 2 above)
Comment (gusty) [reply docsrc/content/abstraction-monad.fsx]: > Also, maybe this should be broken down a little to show the intermediary parts?
It's hard to decipher |> >> >> and <| in one line, even though it is quite neat.

Yes, you have to know all the precedence rules very well to understand it.
Better to use `|>` and `( .. )`.

Comment (NickDarvey) [reply docsrc/content/abstraction-monad.fsx]: > Not sure about using ew here in docs. Is it meaning "either_and_writer" (and ignore the reader bit)?
I assumed so. I built upon the previous example and `ew` was used there.

**Pull request merged**